File: web/src/api/sysOperationRecord.ts
import service from '@/utils/request'

export interface SysOperationRecordSearch {
    page: number
    pageSize: number
    path?: string
    method?: string
    status?: number
}

// @Tags SysOperationRecord
// @Summary 删除SysOperationRecord
export const deleteSysOperationRecord = (data: { ID: number }) => {
    return service({
        url: '/api/v1/operation/record/delete',
        method: 'post',
        data
    })
}

// @Tags SysOperationRecord
// @Summary 删除SysOperationRecord
export const deleteSysOperationRecordByIds = (data: { ids: number[] }) => {
    return service({
        url: '/api/v1/operation/record/delete/multi',
        method: 'post',
        data
    })
}

// @Tags SysOperationRecord
// @Summary 分页获取SysOperationRecord列表
export const getSysOperationRecordList = (params: SysOperationRecordSearch) => {
    return service({
        url: '/api/v1/operation/record/list',
        method: 'get',
        params
    })
}

File: server/router/system/sys_operation_record.go
package system

import (
	"github.com/gin-gonic/gin"
)

type OperationRecordRouter struct{}

func (s *OperationRecordRouter) InitSysOperationRecordRouter(Router *gin.RouterGroup) {
	operationRecordRouter := Router.Group("operation/record")
	{

		operationRecordRouter.POST("delete", operationRecordApi.DeleteSysOperationRecord)            // 删除SysOperationRecord
		operationRecordRouter.POST("delete/multi", operationRecordApi.DeleteSysOperationRecordByIds) // 批量删除SysOperationRecord
		operationRecordRouter.GET("get", operationRecordApi.FindSysOperationRecord)                  // 根据ID获取SysOperationRecord
		operationRecordRouter.GET("list", operationRecordApi.GetSysOperationRecordList)              // 获取SysOperationRecord列表

	}
}
